frontend_app_gallery/src/container/Gallery.jsx

```diff
@@ -67,41 +72,94 @@ export class Gallery extends React.Component {
     addAllResourceI18n(i18n, this.state.config.translation, this.state.loggedUser.lang)
     i18n.changeLanguage(this.state.loggedUser.lang)
 
+    props.registerCustomEventHandlerList([
+      { name: CUSTOM_EVENT.SHOW_APP(this.state.config.slug), handler: this.handleShowApp },
+      { name: CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE, handler: this.handleAllAppChangeLanguage }
+    ])
+
+    props.registerLiveMessageHandlerList([
+      { entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.MODIFIED, optionalSubType: TLM_ST.FILE, handler: this.handleContentCreatedOrModifiedOrUndeleted },
+      { entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.DELETED, optionalSubType: TLM_ST.FILE, handler: this.handleContentDeleted },
+      { entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.UNDELETED, optionalSubType: TLM_ST.FILE, handler: this.handleContentCreatedOrModifiedOrUndeleted }
+    ])
+
     this.lightboxRotation = new LightboxRotation()
+  }
 
-    document.addEventListener(CUSTOM_EVENT.APP_CUSTOM_EVENT_LISTENER, this.customEventReducer)
+  // TLM Handlers
+  handleContentCreatedOrModifiedOrUndeleted = data => {
+    // A picture has been added or modified - let's reload the gallery list.
```

Comment:
if that not handled in this line:

```js
    props.registerLiveMessageHandlerList([
      { entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.MODIFIED, optionalSubType: TLM_ST.FILE, handler: this.handleContentCreatedOrModifiedOrUndeleted }, 
```

with  `optionalSubType`?

Reply:
I'll give you the example of app_html-doc because I think it will be easier to understand. In the app you can create content such as html-document, but you can also create content of the type comment. The optionalSubType is used to differentiate what type of content is being created/modified/deleted/restored, because for specific type of content we may have specific fields that are useless for others (for example, the file type needs the "number of pages" field that is only useful for this type of content).

What I was referring to in the comment is that if you receive a request to delete a file type content, you should check the id of that content before deleting to make sure you are deleting the correct one. If I'm viewing a content X and another user has deleted another content Y, I will receive TLM for that deletion, but it shouldn't influence the content I'm viewing at the moment.

Reply:
I think I will not receive events for comments because of `entityType: TLM_ET.CONTENT`.

For the gallery, we need to update the thumbnails for any image shown, so I cannot just handle the currently displayed picture.

Is this okay or is there something I am missing there?

Reply:
:thinking: I think it's okay then, I hadn't noticed that detail

Reply:
I think you, at least, need to check that the TLM is about the same workspace and the same parent_id as the folder you are opening the gallery on

Reply:
You are right. I did this.